### PR TITLE
ARROW-1849: [GLib] Add input checks to GArrowRecordBatch

### PR DESCRIPTION
**Pull request by kou**

### CODE REVIEW

**Comment by wesm** [comment]
The Go build failed

**Comment by kou** [comment]
Oh, I didn't notice it.
I've fixed.

File: c_glib/arrow-glib/record-batch.cpp [hunk 1]
```diff
@@ -152,7 +171,12 @@ garrow_record_batch_new(GArrowSchema *schema,
   auto arrow_record_batch =
     arrow::RecordBatch::Make(garrow_schema_get_raw(schema),
                              n_rows, arrow_columns);
-  return garrow_record_batch_new_raw(&arrow_record_batch);
+  auto status = arrow_record_batch->Validate();
```
Comment (wesm) [inline c_glib/arrow-glib/record-batch.cpp]: This also checks the schema types match the field types. If that becomes too expensive we might add a faster validation function

**Review by wesm — APPROVED** [review]
+1

File: c_glib/arrow-glib/record-batch.cpp (hunk 1 above)
Comment (kou) [reply c_glib/arrow-glib/record-batch.cpp]: OK. I'll create a JIRA ticket when it's too expensive.

**Pull request closed without merging**